howto/stream/exchange-oob-data.md

```diff
@@ -24,83 +24,111 @@ The following instructions guide you to exchange OOB data using a specific imple
+### Data exchange between Anbox runtime and web client
 
-To send data to an Android application through the channel, the client must use the member function `sendData` of the `AnboxStream` class:
+Launch a stream-enabled instance
 
 ```
-stream.sendData('foo', 'hello world');
+amc launch --name test-app \
+  --raw jammy:android12:amd64 \
+  --enable-streaming
 ```
 
+Once the instance is up and running, retrieve the session id:
 
-### Anbox WebRTC platform
+    amc ls --filter name=test-app --format=csv | awk -F',' '{split($6, r, "="); print r[2]}'
 
-When establishing a peer connection, a number of data channels are set up in the Anbox WebRTC platform on the server side, upon request by the client.
+In the `service.py` file of the stream client sample project, replace `<session_id>` with the session ID retrieved above. Next, launch the web client by opening `https://<appliance_private_ip>:8080/<session_id>` in your browser. Once the web page is fully loaded, it establishes a WebRTC connection to the active session. After the WebRTC peer connection is successfully made, the `foo` data channel will be created on the Anbox runtime server side, in response to the client's request.
 
-At the same time, a number of Unix domain sockets are created under the `/run/user/1000/anbox/sockets` folder. They use the format `webrtc_data_<channel_name>` and represent the established communication bridge between a WebRTC client and the Anbox WebRTC platform. Those Unix domain sockets can be used by a service or daemon to:
+At the same time, a Unix domain socket is created under the `/run/user/1000/anbox/sockets` folder in the format of `webrtc_data_<channel_name>` (`webrtc_data_foo` in the example) and represent the established communication bridge between a WebRTC client and the Anbox runtime. This Unix domain socket can be used by a service or daemon to:
 
 - Receive data sent from a WebRTC client over the data channel and forward it to an Android application.
 - Receive data sent from an Android application and forward it to a WebRTC client over the data channel.
 
-A trivial example to simulate the data transmission between an instance and a WebRTC client is using the [`socat`](https://manpages.ubuntu.com/manpages/bionic/man1/socat.1.html) command to connect the Unix domain socket and perform bidirectional asynchronous data sending and receiving:
+To simulate data transmission between the Anbox runtime and the web client, you can use the [`socat`](https://manpages.ubuntu.com/manpages/bionic/man1/socat.1.html) command to connect the Unix domain socket and perform bidirectional asynchronous data sending and receiving:
 
 1. Connect the Unix domain socket:
 
    ```
    socat - UNIX-CONNECT:/run/user/1000/anbox/sockets/webrtc_data_foo
```

Comment:
Is `socat` installed by default?

Reply:
This correlates with the comment I left above.
https://github.com/canonical/anbox-cloud-docs/pull/228#discussion_r1857709762

We've mentioned [`socat` package](https://manpages.ubuntu.com/manpages/bionic/man1/socat.1.html) above, if people find that socat was not installed, they could easily install that?